### ISSUE TITLE
Weird route choice and insisting behaviour

### ISSUE THREAD

**Issue by Echolon** [organicmaps/organicmaps#7460]
⚠ Have you searched for similar, already existing issues?
Yes, however, I am not sure if this related of any of the existing one like: https://github.com/organicmaps/organicmaps/issues/4290

**Describe the issue**

First of all: Many thanks for this really great app! Navigation was 98% perfect and precise and I enjoyed it a lot!

When I passed by this parking lot, organic maps insisted on taking this route :/

https://www.openstreetmap.org/#map=18/37.71694/-119.65275

**Expected behaviour**
Basically, not choose such a detour. I assume it also is due to weird mapping in OSM so maybe not the apps fault. Maybe some logical changes can prevent it.

**Screenshots**
<img src="https://github.com/organicmaps/organicmaps/assets/29168011/77c82f68-87e5-4a13-a355-8244c2a982cd" alt="drawing" width="400"/>

**System information:**
 - Operating system and its version: [Android 12]
 - Organic Maps version: ?, from October 2023
 - Device Model: Fairphone 4

**Comment by biodranik** [comment]
What is wrong? Did it recommend you to make an u turn?

**Comment by Echolon** [comment]
> What is wrong? Did it recommend you to make an u turn?

I was coming from the rigth turn below and just passing by this parking lot. So, it was not marked as any intermediate spot. The expected route should just have been on the mainroad passing by. But Organic maps was trying to guide me "through" the parking lot.

**Comment by biodranik** [comment]
The screenshot shows an intermediate stop on the route (1). I don't yet understand the issue.

1. Did you build a route with an intermediate stop here?
2. If yes, then was that parking maneuver already in the route preview?
3. If no, then how that intermediate stop appear?

**Comment by Echolon** [comment]
Hey,

I may have used intermediate stops, but certainly not the parking lot.

I tried to reproduce, but I cannot anymore.

Sorry for bothering, can be closed.

24 feb 2024 19:33:49 Alexander Borsuk ***@***.***>:
